### ISSUE TITLE
React | Input issue: readOnly attribute does not work

### ISSUE THREAD

**Issue by rj-david** [OnsenUI/OnsenUI#2570]
__Environment__

  
```
[Core]
  onsenui latest
  
[Framework]
  react 15.4.2

[Framework binding]
 react-onsenui latest

[Platform]
  Desktop - Linux 64-bit

[Browser]
  Desktop - Chrome 70.0.3538.67 on Linux 64-bit
```

__Encountered problem__
  
The `readOnly` attribute does not work. What works is `readonly` (not camel-case)

__How to reproduce__
  
Add `readOnly` as Input attribute

  
  [__Demo link__](https://onsen.io/playground/?issue)

  

- __HTML__

```html
<div id="app"></div>
```

- __JS__

```jsx
var MyPage = React.createClass({
  getInitialState: function() {
    return {
      username: '',
      password: ''
    };
  },
  
  renderToolbar() {
    return (
      <Ons.Toolbar>
      <div className='center'>Form input</div>
      </Ons.Toolbar>
    );
  },

  handleClick: function() {
    if (this.state.username === 'bob' && this.state.password === 'secret') {
      ons.notification.alert('You are now signed in!');
    }
    else {
      ons.notification.alert('Username or password incorrect!');
    }
  },

  handleUsernameChange(e) {
    this.setState({username: e.target.value});
  },

  handlePasswordChange(e) {
    this.setState({password: e.target.value});
  },

  render: function() {
    return (
      <Ons.Page renderToolbar={this.renderToolbar}>
      <section style={{textAlign: 'center'}}>
        <p>
          <Ons.Input
            value={this.state.username}
            onChange={this.handleUsernameChange}
            modifier='underbar'
            float
            readOnly
            placeholder='Username' />
            </p>
            <p>
            <Ons.Input
            value={this.state.password}
            onChange={this.handlePasswordChange}
            modifier='underbar'
            type='password'
            float
            placeholder='Password' />
            </p>
            <p>
            <Ons.Button onClick={this.handleClick}>Sign in</Ons.Button>
            </p>
            </section>
            </Ons.Page>
          );
        }
      });

      ons.ready(function() {
        ReactDOM.render(<MyPage />, document.getElementById('app'));
      });
```

**Comment by emccorson** [comment]
Thanks for reporting this bug. I've confirmed that it is reproducible.

I think the reason `readonly` was working is because React Input automatically passes unknown props to ons-input. ons-input then passes the props to a native input element if they are in [this list](https://github.com/OnsenUI/OnsenUI/blob/4310e6542ac9f02c3e61a41c4c56dd210459ed23/core/src/elements/base/base-input.js#L20):

```  'autocapitalize',
  'autocomplete',
  'autocorrect',
  'autofocus',
  'disabled',
  'inputmode',
  'max',
  'maxlength',
  'min',
  'minlength',
  'name',
  'pattern',
  'placeholder',
  'readonly',
  'required',
  'size',
  'step',
  'validator',
  'value'
```

In other words, all of the items in this list should work as props on the React Input but they are undocumented.

As for `readOnly` (camelCase), it looks like since it is listed as a React prop, it gets converted to "JS style" naming, meaning separated by dashes instead of camelCase i.e. `readOnly` becomes `read-only`.

I'm not sure whether to consider this a documentation problem or a bug. If it's a documentation problem, then the solution is to change `readOnly` to `readonly` in the documentation, and also add all the other missing items from the above list. If it's a bug then I guess `readOnly` should work but `readonly` shouldn't.
